recirq/qcqmc/convert_to_ipie.py

```diff
@@ -0,0 +1,356 @@
+    params = trial_wf_data.params
+    initial_wf = fqe_wfn.Wavefunction([[params.n_elec, 0, params.n_orb]])
+    initial_wf.set_wfn(strategy="hartree-fock")
+    rotated_fqe_wf, unrotated_fqe_wf = optimize_wf.get_evolved_wf(
+        one_body_params=trial_wf_data.one_body_params,
+        two_body_params=trial_wf_data.two_body_params,
+        wf=initial_wf,
+        gate_generators=afqmc_generators.get_pp_plus_gate_generators(
+            n_elec=params.n_elec,
+            heuristic_layers=params.heuristic_layers,
+            do_pp=params.do_pp,
+        ),
+        n_orb=params.n_orb,
+        restricted=params.restricted,
+        initial_orbital_rotation=params.initial_orbital_rotation,
+    )
+    return IPieExportData(
+        rotated_fqe_wf=rotated_fqe_wf,
+        unrotated_fqe_wf=unrotated_fqe_wf,
+        variational_energy=trial_wf_data.ansatz_energy,
+        norm=1.0,
+        path=trial_wf_data.params.base_path.with_suffix(".h5"),
+    )
+
+
+def _get_overlap_export_data(
+    overlap_analysis_data: analysis.OverlapAnalysisData,
+    trial_wf_data: trial_wf.TrialWavefunctionData,
+    hamiltonian_data: hamiltonian.HamiltonianData,
+    threshold: float = 1e-5,
+    k: str = "1",
+) -> IPieExportData:
+    """Get the data to export to ipie from the shadow tomography analysis."""
+    sorted_mapping = list(trial_wf_data.params.mode_qubit_map.items())
+    sorted_mapping.sort(key=lambda x: x[1])
+
+    integer_fermion_qubit_map = {}
+    for i, (old_index, _) in enumerate(sorted_mapping):
+        integer_fermion_qubit_map[old_index.openfermion_standard_index] = i
+
+    reconstructed_wf = np.copy(overlap_analysis_data.reconstructed_wf_for_k[k])
+    reconstructed_wf[np.abs(reconstructed_wf) < threshold] = 0.0
+    norm = float(np.linalg.norm(reconstructed_wf))
+    reconstructed_wf /= norm
+
+    unrotated_fqe_wf = fqe_wfn.Wavefunction(
+        [[trial_wf_data.params.n_elec, 0, trial_wf_data.params.n_orb]]
+    )
+    assert isinstance(unrotated_fqe_wf, fqe_wfn.Wavefunction)
+
+    fqe_conversion.fill_in_wfn_from_cirq(
+        unrotated_fqe_wf, reconstructed_wf, integer_fermion_qubit_map
+    )
+
+    rotated_fqe_wf = rotate_wavefunction_explicitly(
+        unrotated_fqe_wf,
+        trial_wf_data.one_body_basis_change_mat,
+        trial_wf_data.params.n_qubits,
+    )
+
+    unrotated_fqe_wf.scale(norm)
+    rotated_fqe_wf.scale(norm)
+
+    return IPieExportData(
+        rotated_fqe_wf=rotated_fqe_wf,
+        unrotated_fqe_wf=unrotated_fqe_wf,
+        variational_energy=overlap_analysis_data.get_variational_energy(
+            trial_wf_data=trial_wf_data,
+            hamiltonian_data=hamiltonian_data,
+            k=k,
+        ),
+        norm=norm,
+        path=overlap_analysis_data.params.base_path.with_suffix(".h5"),
+    )
+
+
+@attrs.frozen(eq=False)
+class OccCoef:
+    occa: np.ndarray
+    occb: np.ndarray
+    coeffs: np.ndarray
+    """A container for the wavefunction bitstrings and coefficients.
```

Comment:
Typically, the docstring is placed before the class attributes.

Reply:
Whoops, done.